### ISSUE TITLE
Bug Report - Global function reflected to BehaviorContext is not visible in the Script Canvas Node Palette

### ISSUE THREAD

**Issue by AMZN-daimini** [o3de/o3de#8791]
**Describe the bug**
A global function reflected to BehaviorContext in the Reflect function of a new project's SystemComponent doesn't get correctly displayed in the Node Palette of ScriptCanvas.

We verified the reflection code is hit. The bus described in the same function displays correctly, but the global method does not.

**Steps to reproduce**
1. Create a new project (or use an existing one);
2. In the cpp file for the System Component for the new project, define a global function;
3. Reflect the function to BehaviorContext in the Reflect function of the SystemComponent;
4. Build the Editor
5. Search for the function in the Script Canvas Node Palette

**Expected behavior**
The node corresponding to the function shows up and is usable.

**Actual behavior**
The node is not accessible

**Screenshots/Video**
![image](https://user-images.githubusercontent.com/82231674/162340042-d8145074-8db0-4c6c-858e-1c173073fe91.png)

**Found in Branch**
O3DE 21.11.2

**Comment by AMZN-daimini** [comment]
Should try and repro on latest dev before starting an investigation

**Comment by ed420** [comment]
ok i just want to add an update to this.
so continuing to struggle with getting this simple function just to try and get it to work.

![image](https://user-images.githubusercontent.com/13172065/166145012-1c8808d0-282c-4788-aa9f-d52abc9e04dc.png)

so as you see in this image i have declared the same exact function 2 different ways.
one being with a class to start and the second is the same as i did at the original posts bug.
so the first one works perfectly and shows up.
the second one is still having the same bug in script canvas only.
heres an image showing what i mean.

![image](https://user-images.githubusercontent.com/13172065/166145064-f3b83dc3-eaa5-496d-bdce-150957f258f3.png)

i asked around on the discord and it turns out that the first works perfectly for everyone but the second does not.
in fact the second shows up in lua but not scriptcanvas so i can only extrapolate that its something to do with scriptcanvas
directly. hopefully this helps with the bug search.

**Comment by onecent1101** [comment]
For global/free function, you need to add following attribute so ScriptCanvas can pickup
```
->Attribute(AZ::Script::Attributes::Scope, AZ::Script::Attributes::ScopeFlags::Common)
```

Feel free to reopen the issue if it still doesn't work for you.

**Comment by AMZN-daimini** [comment]
> For global/free function, you need to add following attribute so ScriptCanvas can pickup
> 
> ```
> ->Attribute(AZ::Script::Attributes::Scope, AZ::Script::Attributes::ScopeFlags::Common)
> ```
> 
> Feel free to reopen the issue if it still doesn't work for you.

I'm not sure if this actually works as a fix, but even if it does... I'm not sure it should? ScopeFlags default to Launcher, and ScriptCanvas shows Launcher nodes.

**Comment by onecent1101** [comment]
> I'm not sure if this actually works as a fix, but even if it did... I'm not sure it should matter? ScopeFlags default to Launcher, and ScriptCanvas shows Launcher nodes.

It is not a fix, it is current requirement if you want to show global/free function in ScriptCanvas Editor. As this is an Editor use case, it makes sense. Or ScriptCanvas Editor will show bunch of unintended nodes.

**Comment by onecent1101** [comment]
Sounds like you have a feature request for consolidating the usage, feel free to create a separate issue for it.
